controlplane/controlplane/datastore/types/vmmetrics/metric.py

```diff
@@ -0,0 +1,68 @@
+import datetime
+from pydantic import BaseModel
+from typing import Any
+
+
+from sqlalchemy.dialects.postgresql import JSONB
+from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.dialects.postgresql import TIMESTAMP
+from controlplane.datastore.types.base import DatastoreBaseORM
+from controlplane.datastore.types.utils import gen_random_uuid
+from typing import TypeVar, Type, Generic
+
+
+# Define a type variable that can be any subclass of MetricBaseModel
+T1 = TypeVar("T1", bound="MetricBaseModel")
+T2 = TypeVar("T2", bound="MetricLatestBaseModel")
+
+
+class MetricLatestBaseORM(DatastoreBaseORM):
```

Comment:
Add note that this might be pointless now that we codegen stuff instead of trying to solve this via inheritence.